freeipa/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClientBuilder.java

```diff
@@ -138,7 +138,9 @@ public FreeIpaClient build(boolean withPing) throws URISyntaxException, IOExcept
                     .build()) {
                 URI target = getIpaUrl(clientConfig.getApiAddress(), port, basePath, "/session/login_password").toURI();
                 LOGGER.debug("Ping at target: {}", target);
-                client.execute(new HttpHead(target));
+                HttpHead request = new HttpHead(target);
+                additionalHeaders.forEach(request::addHeader);
+                client.execute(request).close();
```

Comment:
you don't have to call the `close()` here, `try` will take care of it

Reply:
I think both the `CloseableHttpResponse` and the `CloseableHttpClient` should be closed.

https://hc.apache.org/httpcomponents-client-ga/tutorial/html/fundamentals.html shows the following:
```
CloseableHttpResponse response = httpclient.execute(httpget);
try {
    <...>
} finally {
    response.close();
}
```

Also `FreeIpaClientBuilder::connect()` does a close on both the `CloseableHttpResponse` and the `CloseableHttpClient`.